__tests__/pages/api/index.test.ts

```diff
@@ -276,310 +291,345 @@ describe('basic OAS scenarios', () => {
-    expect(res.body.oas.reason).toEqual(ResultReasons.YEARS_IN_CANADA)
+    expect(res.body.oas.eligibilityResult).toEqual(ResultKey.CONDITIONAL)
+    expect(res.body.oas.reason).toEqual(ResultReason.YEARS_IN_CANADA)
   })
   it('returns "eligible" when living in No Agreement and 20 years in Canada', async () => {
     const res = await mockGetRequest({
       income: 10000,
       age: 65,
-      livingCountry: LivingCountryOptions.NO_AGREEMENT,
-      legalStatus: LegalStatusOptions.CANADIAN_CITIZEN,
+      livingCountry: LivingCountry.NO_AGREEMENT,
+      legalStatus: LegalStatus.CANADIAN_CITIZEN,
       yearsInCanadaSince18: 20,
     })
-    expect(res.body.oas.eligibilityResult).toEqual(ResultOptions.ELIGIBLE)
-    expect(res.body.oas.reason).toEqual(ResultReasons.NONE)
+    expect(res.body.oas.eligibilityResult).toEqual(ResultKey.ELIGIBLE)
+    expect(res.body.oas.reason).toEqual(ResultReason.NONE)
   })
   it('returns "needs more info" when living in No Agreement and under 20 years in Canada', async () => {
     const res = await mockGetRequest({
       income: 10000,
       age: 65,
-      livingCountry: LivingCountryOptions.NO_AGREEMENT,
-      legalStatus: LegalStatusOptions.CANADIAN_CITIZEN,
+      livingCountry: LivingCountry.NO_AGREEMENT,
+      legalStatus: LegalStatus.CANADIAN_CITIZEN,
       yearsInCanadaSince18: 19,
     })
-    expect(res.body.oas.eligibilityResult).toEqual(ResultOptions.MORE_INFO)
-    expect(res.body.oas.reason).toEqual(ResultReasons.MORE_INFO)
+    expect(res.body.oas.eligibilityResult).toEqual(ResultKey.MORE_INFO)
+    expect(res.body.oas.reason).toEqual(ResultReason.MORE_INFO)
   })
   it('returns "ineligible" when living in No Agreement and under 20 years in Canada and not lived in social country', async () => {
     const res = await mockGetRequest({
       income: 10000,
       age: 65,
-      livingCountry: LivingCountryOptions.NO_AGREEMENT,
-      legalStatus: LegalStatusOptions.CANADIAN_CITIZEN,
+      livingCountry: LivingCountry.NO_AGREEMENT,
+      legalStatus: LegalStatus.CANADIAN_CITIZEN,
       yearsInCanadaSince18: 19,
       everLivedSocialCountry: false,
     })
-    expect(res.body.oas.eligibilityResult).toEqual(ResultOptions.INELIGIBLE)
-    expect(res.body.oas.reason).toEqual(ResultReasons.YEARS_IN_CANADA)
+    expect(res.body.oas.eligibilityResult).toEqual(ResultKey.INELIGIBLE)
+    expect(res.body.oas.reason).toEqual(ResultReason.YEARS_IN_CANADA)
+  })
+  it('returns "ineligible due to age" when 64 and 9 years in Canada and lived in social country', async () => {
+    const res = await mockGetRequest({
+      income: 10000,
+      age: 64,
+      livingCountry: LivingCountry.CANADA,
+      legalStatus: LegalStatus.CANADIAN_CITIZEN,
+      yearsInCanadaSince18: 9,
+      everLivedSocialCountry: true,
+    })
+    expect(res.body.oas.eligibilityResult).toEqual(ResultKey.INELIGIBLE)
+    expect(res.body.oas.reason).toEqual(ResultReason.AGE)
   })
```

Comment:
these are some super thorough tests, good job!

Reply:
yeah, tbh this is my first project with tests, haha! they are a bunch of them but I'm definitely missing some coverage... but is good enough that in general when I break something, the tests catch it, which is the whole point (and I'm really liking working that way)!